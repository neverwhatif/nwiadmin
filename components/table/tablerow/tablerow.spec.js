import React from 'react';
import { shallow } from 'enzyme';

import TableRow from './index';

jest.mock('nwiadmin/components/link', () => 'Link');
jest.mock('nwiadmin/components/reference', () => 'Reference');

const data = {
    $id: 1,
    Title: 'Lorem',
};

const referenceData = {
    $id: 1,
    Title: {
        title: 'Lorem',
        reference: 'IPSUM',
        type: 'reference',
    },
};

const linkData = {
    $id: 1,
    Title: {
        title: 'Lorem',
        path: 'ipsum',
        type: 'link',
    },
};

describe('TableRow', () => {
    const component = shallow(<TableRow data={data} />);
    const componentIsOdd = shallow(<TableRow data={data} isOdd />);
    const componentWithReference = shallow(<TableRow data={referenceData} />);
    const componentWithLink = shallow(<TableRow data={linkData} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render cells', () => {
        expect(component.find('td').length).toBe(1); // Don't include $id
        expect(component.find('td').first().text()).toBe('Lorem');
    });
    it('should render isOdd', () => {
        expect(componentIsOdd.hasClass('rootOdd')).toBe(true);
    });
    it('should render cell with reference', () => {
        expect(componentWithReference.find('Reference').length).toBe(1);
        expect(componentWithReference.find('Reference').prop('title')).toBe('Lorem');
        expect(componentWithReference.find('Reference').prop('reference')).toBe('IPSUM');
    });
    it('should render cell with link', () => {
        expect(componentWithLink.find('Link').length).toBe(1);
        expect(componentWithLink.find('Link').prop('to')).toBe('ipsum');
    });
});
