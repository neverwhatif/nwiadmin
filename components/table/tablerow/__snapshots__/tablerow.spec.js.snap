// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableRow
    data={
      Object {
        "$id": 1,
        "Title": "Lorem",
      }
    }
    functions={null}
    isOdd={false}
    isSelected={false}
    toggleSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        Array [
          <TableCell
            value="Lorem"
          />,
        ],
        undefined,
      ],
      "className": "root",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": "Title",
        "nodeType": "function",
        "props": Object {
          "value": "Lorem",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          Array [
            <TableCell
              value="Lorem"
            />,
          ],
          undefined,
        ],
        "className": "root",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": "Title",
          "nodeType": "function",
          "props": Object {
            "value": "Lorem",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
